.storybook/preview.tsx

```diff
@@ -42,13 +46,35 @@ const withPerseusDecorator: Decorator = (Story) => {
     );
 };
 
-const withThemeSwitcher: Decorator = (Story, context) => {
+const withThemeSwitcher: Decorator = (Story, context: StoryContext) => {
     const theme = context.globals.theme;
+    const [localTheme, setLocalTheme] = React.useState(null);
+    React.useEffect(() => {
+        if (theme) {
+            // Switch the body class based on the theme.
+            document.body.setAttribute(THEME_DATA_ATTRIBUTE, theme);
+            setLocalTheme(theme);
+        }
+    }, [theme]);
+
+    if (context.parameters.enableRenderStateRootDecorator) {
+        return (
+            <RenderStateRoot key={localTheme}>
+                <ThemeSwitcherContext.Provider value={theme}>
+                    <ThemeSwitcher theme={theme}>
+                        <Story />
+                    </ThemeSwitcher>
+                </ThemeSwitcherContext.Provider>
+            </RenderStateRoot>
+        );
+    }
```

Comment:
@jandrade Did I implement this correctly here? In the link you gave, you just highlighted the useEffect, but since it sets a local theme variable, I figured that needed to be included :)

Or maybe the only line we need in the useEffect is `document.body.setAttribute(THEME_DATA_ATTRIBUTE, theme);` ?

Reply:
Removed the if statement and ThemeSwitcherContext